platform/security/idp/security-idp-client/src/test/java/org/codice/ddf/security/idp/client/LogoutRequestServiceTest.java

```diff
@@ -194,10 +196,15 @@ public void testSendLogoutRequestTimeout() throws Exception {
   public void testSendLogoutRequestInvalidNumberOfParams() throws Exception {
     String encryptedNameIdWithTime = nameId + "\n" + time;
     when(encryptionService.decrypt(any(String.class))).thenReturn(nameId);
+    LogoutRequest logoutRequest = mock(LogoutRequest.class);
+    when(logoutRequest.getIssueInstant()).thenReturn(DateTime.now());
+    SessionIndex sessionIndex = mock(SessionIndex.class);
+    when(sessionIndex.getSessionIndex()).thenReturn(SESSION_INDEX);
+    when(logoutRequest.getSessionIndexes()).thenReturn(Collections.singletonList(sessionIndex));
     Response response = logoutRequestService.sendLogoutRequest(encryptedNameIdWithTime);
     assertEquals(Response.Status.SEE_OTHER.getStatusCode(), response.getStatus());
-    String msg =
-        "Failed to decrypt logout request params. Invalid number of params.".replaceAll(" ", "+");
+    insertLogoutRequest();
+    String msg = LogoutRequestService.UNABLE_TO_DECRYPT_LOGOUT_REQUEST.replaceAll(" ", "+");
```

Comment:
Two error messages can mean the same thing, but not be identical. If the strings are checked for identity then even two messages that mean the same thing will fail the test. In that situation, even a spelling correction will make the test fail. It might be better to catch error messages with the wrong meaning in code review.

Reply:
I guess my point is that if those were integers and not Strings, I assume we wouldn't compare against a common constant, but against a value in the test that is independent of the class we're testing against. In other words, would it be better for tests to fail if the strings aren't exactly the same, or test using common constants and miss the fact that a string got mangled by mistake, not find the problem at test time and ship something with the wrong message?

Reply:
I see both sides of the argument. I lean toward constants and taking a risk with mangled strings. You have a broader responsibility ensuring the quality of the app. I will copy the strings as you suggest.